pkg/apis/core/validation/validation.go

```diff
@@ -6329,3 +6335,38 @@ func isHeadlessService(service *core.Service) bool {
 		len(service.Spec.ClusterIPs) == 1 &&
 		service.Spec.ClusterIPs[0] == core.ClusterIPNone
 }
+
+// validateLoadBalancerClassField validation for loadBalancerClass
+func validateLoadBalancerClassField(oldService, service *core.Service) field.ErrorList {
+	allErrs := make(field.ErrorList, 0)
+	if oldService != nil && oldService.Spec.Type == core.ServiceTypeLoadBalancer && service.Spec.Type == core.ServiceTypeLoadBalancer {
```

Comment:
Can;t this be simpler?  

```
if oldService != nil {
  // it's an update
  if oldService.Spec.Type == core.ServiceTypeLoadBalancer && service.Spec.Type == core.ServiceTypeLoadBalancer {
    // old and new are both LoadBalancer
    if (oldService.Spec.LoadBalancerClass == nil) != (service.Spec.LoadBalancerClass == nil) || 
       (oldService.Spec.LoadBalancerClass != nil && *oldService.Spec.LoadBalancerClass != *service.Spec.LoadBalancerClass) {
          // error - can't change class
```

Even better would be some helpers:

```
if isTypeLoadBalancer(oldService) && isTypeLoadBalancer(service) {
    if !sameLoadBalancerClass(oldService, service) {
      // error - can't change class
```

Then it would be readable.